lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -727,72 +837,154 @@ def _labeled_unit_config(
         return static_config
 
 
+def _resolve_dir_against_main_path(*path_elements: str) -> Optional[str]:
+    """Resolve the provided path items against the directory of the main file.
+
+    Look up the directory of the main .py file being executed. This is normally
+    going to be the charm.py file of the charm including this library. Then, resolve
+    the provided path elements and, if the result path exists and is a directory,
+    return its absolute path; otherwise, return `None`.
+    """
+    charm_file = sys.path[0]
+
+    default_alerts_dir = Path(charm_file).joinpath(*path_elements)
+
+    if default_alerts_dir.exists() and default_alerts_dir.is_dir:
+        return str(default_alerts_dir.absolute())
+
+    return None
+
+
 class MetricsEndpointProvider(Object):
     """Construct a metrics provider for a Prometheus charm."""
 
     def __init__(
         self,
         charm,
-        name,
-        service_event,
+        relation_name: str = DEFAULT_RELATION_NAME,
         jobs=[],
-        alert_rules_path="src/prometheus_alert_rules",
+        alert_rules_path: Optional[str] = None,
```

Comment:
Again there should be no need of `Optional[str]`. This can just be an empty string by default.